packages/frontpage/app/(app)/moderation/page.tsx

```diff
@@ -0,0 +1,200 @@
+"use server";
+import { Shield } from "lucide-react";
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from "@/lib/components/ui/card";
+import { ensureUser, isAdmin as isAdmin } from "@/lib/data/user";
+import { redirect } from "next/navigation";
+import {
+  getModeratorReportStats,
+  getReport,
+  getReports,
+  updateReport,
+} from "@/lib/data/db/report";
+import {
+  ModerationEventDTO,
+  createModerationEvent,
+} from "@/lib/data/db/moderation";
+import { PostCollection } from "@/lib/data/atproto/post";
+import { CommentCollection } from "@/lib/data/atproto/comment";
+import { revalidatePath } from "next/cache";
+import Link from "next/link";
+import { ReportCard } from "./_components/report-card";
+import { moderatePost } from "@/lib/data/db/post";
+import { DID } from "@/lib/data/atproto/did";
+import { moderateComment } from "@/lib/data/db/comment";
+import { moderateUser } from "@/lib/data/db/user";
+
+export async function performModerationAction(
+  input: { reportId: number; status: "accepted" | "rejected" },
+  _: FormData,
+) {
+  "use server";
+  const user = await ensureUser();
+  const report = await getReport(input.reportId);
+
+  if (!report) {
+    throw new Error("Report not found");
+  }
+
+  const newModEvent: ModerationEventDTO = {
+    subjectUri: report.subjectUri,
+    subjectDid: report.subjectDid,
+    createdBy: user.did,
+    createdAt: new Date(),
+    labelsAdded: report.reportReason,
+    creatorReportReason: report.creatorComment,
+  };
+
+  if (report.subjectCollection) {
+    if (report.subjectCollection === PostCollection) {
+      newModEvent.subjectCollection = PostCollection;
+    } else if (report.subjectCollection === CommentCollection) {
+      newModEvent.subjectCollection = CommentCollection;
+    }
+
+    newModEvent.subjectRkey = report.subjectRkey;
+    newModEvent.subjectCid = report.subjectCid;
+  }
+
+  const modAction = async () => {
+    switch (report.subjectCollection) {
+      case PostCollection:
+        return await moderatePost({
+          rkey: report.subjectRkey!,
+          authorDid: report.subjectDid! as DID,
+          cid: report.subjectCid!,
+          hide: input.status === "accepted",
+        });
+
+      case CommentCollection:
+        return await moderateComment({
+          rkey: report.subjectRkey!,
+          authorDid: report.subjectDid! as DID,
+          cid: report.subjectCid!,
+          hide: input.status === "accepted",
+        });
+
+      default:
+        return await moderateUser({
+          userDid: report.subjectDid as DID,
+          hide: input.status === "accepted",
+          label: report.reportReason ?? "",
```

Comment:
Label is optional, but we're providing a default empty string value which is kinda confusing. Is that empty string load bearing?

Would clear things up by changing the type of the `label` option to be `string | null` and then simply passing `label: report.reportReason,`